app/src/main/java/de/westnordost/streetcomplete/quests/barrier_type/AddStileType.kt

```diff
@@ -0,0 +1,46 @@
+package de.westnordost.streetcomplete.quests.barrier_type
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.osm.osmquest.OsmFilterQuestType
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder
+
+class AddStileType : OsmFilterQuestType<BarrierType>() {
+
+    override val elementFilter = """
+        nodes with barrier=stile and !stile
+    """
```

Comment:
exclude private barriers?

Reply:
yes, no, perhaps?

Reply:
there are some with access=private/no but with foot=designated - https://www.openstreetmap.org/node/5000261512

I found one without extra tags ( https://www.openstreetmap.org/node/3978247540 ), but looks like a tagging mistake https://www.openstreetmap.org/note/2632723

`foot=yes` `foot=permissive` also sometimes amend `access`

And even for private stiles asking still is not harmful.

Though https://www.openstreetmap.org/node/7450489924 https://www.openstreetmap.org/node/891091396 https://www.openstreetmap.org/node/6034850648 https://www.openstreetmap.org/node/3707227051 seem actually private

But it is just tiny, tiny fraction of all. https://overpass-turbo.eu/s/16rU

------------------

So if added filter would block on `access=private` `access=no` and let `foot=yes` `foot=permissive` `foot=designated` to override it.

Reply:
Actually I meant access on the way that crosses the stile

Reply:
I will check but likely the same applies, especially in UK such rights of way are common (path on a private land with right of way for pedestrians)

Reply:
We have `access !~ private|no` for many quests

Reply:
done, tested